src/e2e-test/java/io/cdap/plugin/odp/stepsdesign/RuntimeODP.java

```diff
@@ -0,0 +1,422 @@
+    CdfBigQueryPropertiesLocators.datasetProjectID.sendKeys(CDAPUtils.getPluginProp("odpProjectId"));
+    CdfBigQueryPropertiesLocators.dataset.sendKeys(CDAPUtils.getPluginProp("dataSetOdp"));
+    CdfBigQueryPropertiesLocators.bigQueryTable.sendKeys(CDAPUtils.getPluginProp(tableName));
+    CdfBigQueryPropertiesLocators.truncatableSwitch.click();
+    CdfBigQueryPropertiesLocators.updateTable.click();
+    CdfBigQueryPropertiesLocators.validateBttn.click();
+    SeleniumHelper.waitElementIsVisible(CdfBigQueryPropertiesLocators.textSuccess, 1L);
+  }
+
+  @Then("Close the BQ Properties")
+  public void closeTheBQProperties() {
+    CdfGcsActions.closeButton();
+  }
+
+  @Then("Run the ODP Pipeline in Runtime")
+  public void runTheODPPipelineInRuntime() throws InterruptedException {
+    CdfPipelineRunAction.runClick();
+  }
+
+  @Then("Wait till ODP pipeline is in running state")
+  public void waitTillODPPipelineIsInRunningState() {
+    WebDriverWait wait = new WebDriverWait(SeleniumDriver.getDriver(), 1000000);
+    wait.until(ExpectedConditions.or(
+      ExpectedConditions.visibilityOfElementLocated(By.xpath("//*[@data-cy='Succeeded']")),
+      ExpectedConditions.visibilityOfElementLocated(By.xpath("//*[@data-cy='Failed']"))));
+  }
+
+  @Then("Open Logs of ODP Pipeline")
+  public void openLogsOfODPPipeline() throws InterruptedException, FileNotFoundException {
+    CdfPipelineRunAction.logsClick();
+    Thread.sleep(5000); //TODO
+    rawLog = CdfPipelineRunAction.captureRawLogs();
+    BeforeActions.scenario.write(rawLog);
+    out.println(rawLog);
+    out.close();
+  }
+
+  @Then("Verify the ODP pipeline status is {string}")
+  public void verifyTheODPPipelineStatusIs(String status) {
+    boolean webelement = SeleniumHelper.verifyElementPresent("//*[@data-cy='" + status + "']");
+    Assert.assertTrue(webelement);
+  }
+
+  @Then("validate successMessage is displayed for the ODP pipeline")
+  public void validateSuccessMessageIsDisplayedForTheODPPipeline() {
+    CdfLogActions.validateSucceeded();
+  }
+
+  @Then("Get Count of no of records transferred from ODP to BigQuery in {string}")
+  public void getCountOfNoOfRecordsTransferredFromODPToBigQueryIn(String table) throws
+    IOException, InterruptedException {
+    countRecords = gcpClient.countBqQuery(CDAPUtils.getPluginProp(table));
+  }
+
+  @Then("Verify the Delta load transfer is successful in {string} on basis of {string}")
+  public void verifyTheDeltaLoadTransferIsSuccessfull(String table, String field) throws IOException,
+    InterruptedException {
+    int i = 0;
+    String projectId = CDAPUtils.getPluginProp("odpProjectId");
+    String datasetName = CDAPUtils.getPluginProp("dataSetOdp");
+    String selectQuery = "SELECT count(*)  FROM `" + projectId + "." + datasetName + "." + CDAPUtils.getPluginProp
+      (table) + "` WHERE " +
+      field.toUpperCase();
+    switch (datasource) {
+      case "0MATERIAL_ATTR":
```

Comment:
use constants for datasources and actions if possible

Reply:
done